slack/src/faceswapbot/faceswapbot.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/nlopes/slack"
+	"log"
+	"os"
+	"strings"
+	"reflect"
+	"net/http"
+	"time"
+	"io/ioutil"
+	"os/exec"
+	"bytes"
+)
+
+var (
+	botToken string
+	faceSwapExec string
+)
+
+func main() {
+	var (
+		isDebug  bool
+	)
+
+	flag.StringVar(&botToken, "token", "", "Your SlackBot Token")
+	flag.BoolVar(&isDebug, "debug", false, "Debug")
+	flag.Parse()
+
+	if botToken == "" {
+		fmt.Println("Slack SlackBot token cannot be empty")
+		return
+	}
+
+	fileTypes := []string{"jpg", "png"}
+
+	api := slack.New(botToken)
+	logger := log.New(os.Stdout, "slack-bot: ", log.Lshortfile|log.LstdFlags)
+	slack.SetLogger(logger)
+	api.SetDebug(isDebug)
+
+	rtm := api.NewRTM()
+	go rtm.ManageConnection()
+
+	for msg := range rtm.IncomingEvents {
+		//fmt.Print("Event Received: ")
+		switch ev := msg.Data.(type) {
+		case *slack.MessageEvent:
+			info := rtm.GetInfo()
+			botId := fmt.Sprintf("<@%s>", info.User.ID)
+
+			if ev.User != info.User.ID && strings.Contains(ev.Text, botId) && ev.SubType == "file_share" {
+				isFileAllowed, _ := inArray(ev.File.Filetype, fileTypes)
+
+				if isFileAllowed {
+					// Handle the file
+					handleFile(rtm, ev)
+				}
+			}
+
+		case *slack.RTMError:
+			fmt.Printf("Error: %s\n", ev.Error())
```

Comment:
Should there be a break after this and the previous case?

Reply:
No, see here https://github.com/golang/go/wiki/Switch#break and https://github.com/golang/go/wiki/Switch#fall-through

Reply:
Aha, something new! I guess it makes sense, fall through is the most common thing to do.